### PR TITLE
pdctl: support dynamic addition of witness-related schedulers

### LINKED ISSUE #5762: add balance witness scheduler
## Development Task

ref https://github.com/tikv/tikv/issues/12876

### PR DESCRIPTION
**Pull request by ethercflow**
ref tikv/tikv#12876

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5762 

### What is changed and how does it work?

```commit-message
pdctl: support dynamic addition of witness-related schedulers
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

Code changes

- pdctl 

Side effects

- None

Related changes

- None

### Release note

```release-note
None
```

### CODE REVIEW

**Review by rleungx — COMMENTED** [review]
LGTM, please add a test.

**Comment by ethercflow** [comment]
> LGTM, please add a test.

Added, thanks!

**Review by rleungx — APPROVED** [review]

File: tools/pd-ctl/pdctl/command/scheduler.go [hunk 1]
```diff
@@ -352,6 +354,38 @@ func NewGrantHotRegionSchedulerCommand() *cobra.Command {
...
+		Run:   addSchedulerForBalanceWitnessCommandFunc,
+	}
+	return c
+}
+
+// NewTransferWitnessLeaderSchedulerCommand returns a command to add a transfer-witness-leader-shceudler.
+func NewTransferWitnessLeaderSchedulerCommand() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "transfer-witness-leader-scheduler",
+		Short: "add a scheduler to transfer witness leader",
```
Comment (bufferflies) [inline tools/pd-ctl/pdctl/command/scheduler.go:371]:
```suggestion
		Short: "add a scheduler to transfer witness peer",
```

File: server/api/scheduler.go [hunk 2]
```diff
@@ -160,6 +160,11 @@ func (h *schedulerHandler) CreateScheduler(w http.ResponseWriter, r *http.Reques
 			h.r.JSON(w, http.StatusInternalServerError, err.Error())
 			return
 		}
+	case schedulers.TransferWitnessLeaderName:
```
Comment (bufferflies) [inline server/api/scheduler.go:163]: Does it need to handle the balance-witness-scheduler type?

Comment (ethercflow) [reply server/api/scheduler.go:163]: Yes, but already handled.

**Review by bufferflies — APPROVED** [review]

**Pull request merged**